network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/RatioTapChangerAdderImpl.java

```diff
@@ -155,6 +160,11 @@ public RatioTapChanger add() {
                     + tapPosition + " [" + lowTapPosition + ", "
                     + highTapPosition + "]");
         }
+        if (solvedTapPosition != null && (solvedTapPosition < lowTapPosition || solvedTapPosition > highTapPosition)) {
```

Comment:
You should implement a static check method. You have this code in 3 classes

Reply:
checkSolvedTapPositionCreation() and checkSolvedTapPositionCreation() could be one method. With a boolean isSolvedTapPosition for the log

Reply:
done